doc/index.md

```diff
@@ -236,114 +292,131 @@ local myconnection = http_client.connect {
-
-### `new(socket, conn_type, version)` <!-- --> {#http.h1_connection.new}
-
-
-### `h1_connection.version` <!-- --> {#http.h1_connection.version}
-
-Either `1.0` or `1.1`
+  - `options` is a table containing:
 
+	  - `tls` (boolean|userdata, optional): the `SSL_CTX*` to use, or a boolean to indicate the default TLS context.  
+		defaults to `true`.
 
-### `h1_connection:connect(timeout)` <!-- --> {#http.h1_connection:connect}
+		  - `true` indicates to use the default TLS settings, see [*http.tls*](#http.tls) for information.
+		  - `false` means do not negotiate TLS
 
-See [`connection:connect(timeout)`](#connection:connect)
+	  - `version` (nil|1.0|1.1|2): HTTP version to use.
+		  - `nil`: attempts HTTP 2 and falls back to HTTP 1.1
+		  - `1.0`
+		  - `1.1`
+		  - `2`
 
+	  - `h2_settings` (table, optional): HTTP 2 settings to use. See [*http.h2_connection*](#http.h2_connection) for details
 
-### `h1_connection:checktls()` <!-- --> {#http.h1_connection:checktls}
 
-See [`connection:checktls()`](#connection:checktls)
+## http.h1_connection
 
+The h1_connection module adheres to the [connection](#connection) and provides HTTP 1 and 1.1 specific operations.  
 
-### `h1_connection:localname()` <!-- --> {#http.h1_connection:localname}
+### `connection.new(socket, conn_type, version)` <!-- --> {#connection.new}
 
-See [`connection:localname()`](#connection:localname)
+Constructor for a new connection. Takes a socket instance, a connection type string and a numeric HTTP version number. Valid values for the connection type are "client" and "server". Valid values for the version number are 1 and 1.1. Success returns the newly initialized connection object in a non-connected state. Failure returns 
 
 
-### `h1_connection:peername()` <!-- --> {#http.h1_connection:peername}
+### `h1_connection.version` <!-- --> {#http.h1_connection.version}
 
-See [`connection:peername()`](#connection:peername)
+Used to specificy the http version used for the connection handshake. Valid values are:
+- `1.0` 
+- `1.1`
 
 
 ### `h1_connection:clearerr(...)` <!-- --> {#http.h1_connection:clearerr}
 
+ 
 
 ### `h1_connection:take_socket()` <!-- --> {#http.h1_connection:take_socket}
 
 
-### `h1_connection:shutdown(dir)` <!-- --> {#http.h1_connection:shutdown}
-
-See [`connection:shutdown()`](#connection:shutdown)
-
-
-### `h1_connection:close()` <!-- --> {#http.h1_connection:close}
-
-See [`connection:close()`](#connection:close)
-
-
-### `h1_connection:new_stream()` <!-- --> {#http.h1_connection:new_stream}
-
-See [`connection:new_stream()`](#connection:new_stream)
-
-
-### `h1_connection:get_next_incoming_stream(timeout)` <!-- --> {#http.h1_connection:get_next_incoming_stream}
-
-See [`connection:get_next_incoming_stream()`](#connection:get_next_incoming_stream)
-
-
 ### `h1_connection:flush(...)` <!-- --> {#http.h1_connection:flush}
 
+Flushes all buffered data on the socket. Returns true on success. Returns false and the error if the socket fails to flush.
```

Comment:
Should return nil, error_message, errno on failure (and hence not need to be mentioned)